README.md
<p align="center">
  <aa href="https://www.magicllight.com">
    <img src="magicllight.png?raw=true" width="90%"/>
  </a></p>

#  <pre>✨ MagicLLight (MLL) <br> &nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;Experience the Future<p align="center">🞉</a></p></pre>
##### (Landing Soon)
#### Works with: ✰ Ruby.ai <> @ openai <> ● open-interpreter <> 火 Akyeo <> 🧠 Claude2 <> 🦜 Langchain <> 🎙 ElevenLabs
### Features: 
* Live Interaction on your Desktop,
* Realtime Voice to Voice **in Any Language!**
* Easy As Pie - Custom Embed AI
	* for your own website
* Works on any webpage (using extension)
* **Whatsapp & Telegram Sync**
* Awesome on Mobile 

	will soon support:
	litellm (any llm) <> atomai <> EmployeeGPT


## Magically Sync With your Life
	
### Plans & Pricing:
* Dev/Hobbiest:
	* Freely use forever, bring your own models
* For Personal Use:
	*  use your openai/claude/ruby api key, or choose one of the our models (10-30$ month or pay as you go)
* Premium User:
	* All features out of the box, private files, multimodel, agents & much more (read here)
* Small Business:
	* Use your own branding, private knowledge on your site (Embed), 
* Entreprise:
	* Commercial License Can be recived from us based on approval
  
    
##### To join our Private Beta: *Click here* - Be first to test the cutting edge



# 💻 Developers:

### Coming soon:
* Easy Install for Mac/Windows/Linux
* Easy Web Embed
* Lazy/Fast Cached Responses and Quick code execution
* Branching History
* RLHF support
* Auto Agent & Context Chooser
* Grandma Mode: Work alongside technologically challenged/inexperienced users
* Press Capslock for live, continues voicechat mode
* Enhanced Translations
* Realtime audio response
* voiceSlicer - take chunks stream of "content" , process sections, with expanding fallback, generate audio files, send them to play queue
* Multi windows, render images



### Showcasing:
* If you made a cool feature [click here] to apply for showcase
* If you have a cool idea [click here]
* More projects coming soon

Visit & Rate our *Idea Board* (signin with github)<br>
*Discord* : [link] <br>
*Status* : Currently not engaging to let us focus on building, free from distractions,<br>
 &nbsp;&nbsp;  &nbsp;&nbsp; &nbsp;&nbsp; &nbsp;&nbsp;&nbsp;&nbsp;You're welcome to join community chat and put gh prs in our inbox. we'll keep our eyes out 


<p align="center">
  <a href="https://www.wholesomegarden.com">
    <img src="ezgif-2-e085578e27.gif?raw=true" width="200"/>
  </a>
<br><bold>Be Powered by ✨ MagicLLight</bold>

</p>


pycookiecheat/CHANGELOG.md
# [Changelog](https://keepachangelog.com)

## v0.5.0 :: 20230324

- Add support for Brave thanks to @chrisgavin!
- Add support for Slack thanks to @hraftery!
- Migrate config to pyproject.toml alone
- Minor cleanup to codebase and tests

## v0.4.7 :: 20210826

- No noteworthy API changes, hence the bugfix version bump, but some major
  infrastructure and testing updates:
    - Now uses GitHub Actions instead of Travis
    - Now uses Playwright for testing, to actually open a Chromium instance and
    use a real `Cookies` database
    - PEP517
    - black
- Now requires python >= 3.7
    - This is largely due to requiremets of Playwright:
      https://pypi.org/project/playwright/, which is only a *test* dependency
    - Because I can't *test* with <=3.6, I'm not listing it as compatible,
      though it *probably* will still work
- Migrate to pyproject.toml

## v0.4.6 :: 2019111

- Try to open Chrome database in read-only mode to avoid db locked errors (#29)

## v0.4.5 :: 20191007

- db6ac6d Go back to using cryptography due to
  https://www.cvedetails.com/cve/CVE-2013-7459/
- c70ad51 Allow users to override password (thanks @alairock)

## v0.4.4 :: 20180706

- Optionally outputs cookies to a file compatible with cURL (thanks to
  Muntashir Al-Islam!)

## v0.4.3 :: 20170627

- Consistently use Chrome as default across platforms, allow user to specify
  Chromium as desired (thanks @jtbraun)

## v0.4.0 :: 20170504

- Remove compatibility for Python <3.5
- Add type hints
- Refactor for smaller functions
- Expand docstrings
- Revert from `cryptography` back to `PyCrypto` and `hashlib` for easier
  installation.

## v0.3.4 :: 20170414

- Add support for new Ubuntu keyring / libsecret
  - See <https://github.com/n8henrie/pycookiecheat/issues/12> for details
  - Many thanks to @stat1c1c3au and @trideceth12 for contributions

## 0.3.0

- Use [`cryptography`](https://cryptography.io/en/latest/) instead of
  `pycrypto` (thanks to [Taik](https://github.com/Taik)!)
  - Seems to be [significantly
    faster](https://github.com/n8henrie/pycookiecheat/pull/11#issuecomment-221950400)
  - Works with PyPy >= 2.6.0 (no support for PyPy3 yet)

## 0.2.0

- Fix domain and subdomain matching
- Make SQL query more secure by avoiding string formatting
- Many thanks to [Brandon Rhodes](https://github.com/brandon-rhodes) for 24c4234 !

## 0.1.10

- Read version to separate file so it can be imported in setup.py
- Bugfix for python2 on linux

## 0.1.9

- Bugfix for python2 on linux

## 0.1.8

- Python2 support (thanks [dani14-96](https://github.com/dani14-96))

## 0.1.7

- Configurable cookies file (thanks [ankostis](https://github.com/ankostis))

## 0.1.6

- OSError instead of Exception for wrong OS.
- Moved testing requirements to tox and travis-ci files.

## 0.1.5

- Updated to work better with PyPI's lack of markdown support
- Working on tox and travis-ci integration
- Added a few basic tests that should pass if one has Chrome installed and has visited my site (n8henrie.com)
- Added sys.exit(0) if cookie_file not found so tests pass on travis-ci.

## 0.1.0 (2015-02-25)

- First release on PyPI.

## Prior changelog from Gist

- 20150221 v2.0.1: Now should find cookies for base domain and all subs.
- 20140518 v2.0: Now works with Chrome's new encrypted cookies.


transparwnt-web-app/weather.py
from selenium import webdriver
from selenium.webdriver.chrome.options import Options
from bs4 import BeautifulSoup
import requests
options = Options() 
options.add_argument("--headless")

driver = webdriver.Chrome(options=options)
lang="iw"
url = 'https://www.google.com/search?q=%D7%9E%D7%96%D7%92+%D7%90%D7%95%D7%95%D7%99%D7%A8+%D7%94%D7%A8%D7%A6%D7%9C%D7%99%D7%94'
url = 'https://www.google.com/search?q=is+it+going+to+rain+tomorrow+&hl=iw'
url = 'https://www.google.com/search?q=is+it+going+to+rain+tomorrow+&hl=iw'
url = 'https://www.google.com/search?q=what+is+mount+everest&hl=iw'
url = 'https://www.google.com/search?q=what+is+a+dog&oq=what+is+a+dog&h1=iw'
url = 'https://www.google.com/search?q=what+is+a+dog&h1=iw'
url = 'https://www.google.com/search?q=cat&h1=iw'


driver.get(url)

# Wait for page to fully render
driver.implicitly_wait(.8)  

soup = BeautifulSoup(driver.page_source, 'html.parser')

# Find the results div
results_div = soup.find('div', {'id': 'rso'})
#weather = soup.find('div', {'id': 'rso'})
weather = soup.find('div', {'data-ve-view': True})

forcast = []


def extract_last_div_with_attributes():
    # Find the last <div> element with attributes aria-level="2" and role="heading"
    last_div_element = None
    for div in reversed(soup.find_all('div', {'aria-level': '2', 'role': 'heading'})):
        last_div_element = div
        break
    c = 0
    if last_div_element:
        parent = last_div_element.parent
        while len(parent.find_all(recursive=False)) == 1 or c < 3:
            c+=1
            print(":::::::::::::",c)
            parent = parent.parent
        print(":::::::::::::xxx",c)

        return parent
        # Strip empty texts and combine all texts with "\n"
        text = "\n".join(filter(None, (x.strip() for x in parent.stripped_strings)))

        return text

    return None
    
extracted_text = extract_last_div_with_attributes()

if extracted_text:
    # Save the extracted text as 'info_v2.html'
    with open('info_v2.html', 'w', encoding='utf-8') as file:
        file.write(extracted_text.prettify())
    print("Saved 'info_v2.html' with extracted text.")
else:
    print("No suitable element found with the specified attributes.")    

print("DONE!")
exit()

def extract_info_from_html():
    result = {}

    # Parse the HTML content
    #soup = BeautifulSoup(html_content, 'html.parser')

    # Find the element with attribute data-attrid="title"
    title_element = soup.find('h2', {'data-attrid': 'title'})
    if title_element:
        result['title'] = title_element.text

    # Find the index of the last <h3> element on the page
    h3_elements = soup.find_all('h3')
    if h3_elements:
        last_h3_index = len(h3_elements) - 1

        # Find the text values of the next two <span> elements
        span_elements = soup.find_all('span')
        next_span_texts = []
        for i in range(last_h3_index + 1, last_h3_index + 3):
            if i < len(span_elements):
                next_span_texts.append(span_elements[i].text)

        if len(next_span_texts) == 2:
            result['description'] = next_span_texts[0]
            result['link'] = next_span_texts[1]

        # Find the index of the next <span> element
        next_span_index = last_h3_index + 3
        if next_span_index < len(span_elements):

            # Find the text content of all <div> elements between the two <span> elements and the span after them
            div_elements_between_spans = []
            for i in range(next_span_index, len(span_elements)):
                if span_elements[i].name == 'span':
                    break
                elif span_elements[i].name == 'div':
                    div_elements_between_spans.append(span_elements[i].text)

            result['extraInfoList'] = div_elements_between_spans

    return result



def find_last_h2_parent():
    # Parse the HTML content
    #soup = BeautifulSoup(html_content, 'html.parser')

    # Find the last <h2> element on the page and its index
    h2_elements = soup.find_all('h2')
    if h2_elements:
        last_h2_index = len(h2_elements) - 1
        last_h2_element = h2_elements[last_h2_index]

        # Walk backward through parent elements
        parent = last_h2_element
        skip = 6 + 2 #(+2 for margin, can be up to 4 or more)
        layers = 0
        lookfor = "גובה"
        while parent:
            if len(parent.find_all(recursive=False)) > 1 and (skip == 0):# or lookfor in parent.text):
                # If the parent has more than one child, return it prettified
                print("$$$$$$$$$$$$$$$$$$$$$$$$$$$$ FOUND")
                print("$$$$$$$$$$$$$$$$$$$$$$$$$$$$ FOUND")
                print("$$$$$$$$$$$$$$$$$$$$$$$$$$$$ FOUND")
                print("$$$$$$$$$$$$$$$$$$$$$$$$$$$$ FOUND")
                print("$$$$$$$$$$$$$$$$$$$$$$$$$$$$ FOUND",layers,skip)
                return parent
            elif skip>0:
                layers += 1
                skip-=1
            parent = parent.parent

    return None  # Return None if no suitable parent is found

if weather:
    #results_html = results_div.prettify()
    results_html = weather.prettify()

    # Extract data here
    #print(results_html)
    results_html = weather.prettify()

    # Create a complete HTML document with head and body tags
    full_html = f"<!DOCTYPE html>\n<html>\n<head>\n"

    # Extract and include CSS links from the original HTML
    css_links = soup.find_all('link', rel='stylesheet')
    for link in css_links:
        css_url = link.get('href')
        if css_url and not css_url.startswith(('http:', 'https:')):
            # Make the CSS URL absolute by joining it with the base URL of the page
            base_url = 'https://www.google.com'  # Replace with the actual base URL
            css_url = urljoin(base_url, css_url)
            ##css_url = urljoin(url, css_url)


        # Fetch the CSS content
        css_response = requests.get(css_url)
        if css_response.status_code == 200:
            css_content = css_response.text

            # Include the CSS content in the HTML document
            #full_html += "<style>.boxx{width:400;height:400;positon:absolute;}"+f"{css_content}</style>"

    full_html += f"\n</head>\n<body><div class='boxx'>\n{results_html}\n</div></body>\n</html>"
    # Extract all aria-label attributes and print their values
    aria_label_values = [element.get('aria-label') for element in soup.find_all(attrs={'aria-label': True})]
    
    for value in aria_label_values:
        if value and "°" in value and value.split(" ")>1:
            print(value)
            forcast.append(value)
    # Save the HTML to a file
    with open('weather2.html', 'w', encoding='utf-8') as file:
        file.write(full_html)
    print("DONE!")


info = extract_info_from_html()

print("%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%\n",info)
res = find_last_h2_parent()
print("%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%\n",res)
prettified_element=res
if prettified_element:
 # Get the original CSS styles by extracting and including CSS links in the head
    #css_links = soup.find_all('link', rel='stylesheet')
    #css_styles = [link.get('href') for link in css_links if link.get('href')]

    # Include the CSS links in the head of the prettified element
    head = soup.new_tag('head')
    #for css_style in css_styles:
        #link_tag = soup.new_tag('link', href=css_style, rel='stylesheet')
        #head.append(link_tag)
        
    prettified_element.insert_before(head)

    # Create a complete HTML document with html and body tags
    full_html = f"<!DOCTYPE html>\n<html>\n{prettified_element}\n</html>"

    # Save the HTML to a file
    with open('info.html', 'w', encoding='utf-8') as file:
        file.write(full_html)
    print("Saved 'info.html' with original CSS styles.")
    
    # Trim 'info.html' by removing content below the first element with aria-level
    soup = BeautifulSoup(full_html, 'html.parser')
    first_aria_level_element = soup.find(attrs={'aria-level': True})
    if first_aria_level_element:
        for tag in first_aria_level_element.find_all_next():
            tag.decompose()
            

    # Save the trimmed HTML to 'info.html'
    with open('info.htmlD', 'w', encoding='utf-8') as file:
        file.write(soup.prettify())
    print("Trimmed 'info.html' by removing content below the first element with aria-level.")

        
    # Filter out empty or whitespace-only <div> elements
    for div_element in soup.find_all('div'):
        if not div_element.text.strip():
            div_element.extract()

    # Save the trimmed HTML to 'info.html'
    with open('info_raw_dataD.html', 'w', encoding='utf-8') as file:
        file.write(soup.prettify())
    print("saved raw content")



print("FORCAST:")
for f in  forcast:
    print(f)



driver.quit()
